modules/javafx.graphics/src/main/native-glass/mac/GlassViewDelegate.m

```diff
@@ -845,6 +851,72 @@ - (void)sendJavaGestureEndEvent:(NSEvent *)theEvent
     GLASS_CHECK_EXCEPTION(env);
 }
 
+/*
+ * This method is a replacement for the deprecated beginGestureWithEvent
+ * method, which is no longer delivered to a View by macOS. This
+ * is called for each gesture event to track the beginning of a
+ * gesture using the phase of the event. We call sendJavaGestureBeginEvent
+ * if there are no other gestures active.
+ */
+- (void)maybeBeginGestureWithEvent:(NSEvent *)theEvent withMask:(GestureMaskType)theMask
+{
+    NSEventPhase phase = [theEvent phase];
+    if (phase == NSEventPhaseBegan) {
+        if (gesturesBeganMask == 0) {
+            [self sendJavaGestureBeginEvent:theEvent];
+        }
+        gesturesBeganMask |= theMask;
+    }
+}
+
+/*
+ * This method is a replacement for the deprecated endGestureWithEvent
+ * method, which is no longer delivered to a View by macOS. This
+ * is called for each gesture event to track the end of a
+ * gesture using the phase of the event. We call sendJavaGestureEndEvent
+ * if there are no other gestures active.
+ */
+- (void)maybeEndGestureWithEvent:(NSEvent *)theEvent withMask:(GestureMaskType)theMask
+{
+    NSEventPhase phase = [theEvent phase];
+    if (phase == NSEventPhaseEnded || phase == NSEventPhaseCancelled) {
+        if ((gesturesBeganMask & theMask) != 0) {
+            gesturesBeganMask &= ~theMask;
+            if (gesturesBeganMask == 0) {
+                [self sendJavaGestureEndEvent:theEvent];
+            }
+        }
```

Comment:
In continuation to the change in `maybeBeginGestureWithEvent`, we would need to remove the inner `if` condition change here,

```
if ((gesturesBeganMask & theMask) != 0) {
    gesturesBeganMask &= ~theMask;
    [self sendJavaGestureEndEvent:theEvent];
}

```